src/auto/AutoCriterion.lua

```diff
@@ -0,0 +1,26 @@
+local AutoCriterion, parent = torch.class('nn.AutoCriterion', 'nn.Criterion')
```

Comment:
Hmmm, fair point, I suppose for now there's no good out of the box solution for that. @andresy introduced this recently FYI: https://github.com/torch/torch7/tree/master/lib/luaT#const-char-luat_newlocalmetatablelua_state-l-const-char-tname-const-char-parenttname-lua_cfunction-constructor-lua_cfunction-destructor-lua_cfunction-factory-int-moduleidx